hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/protocolPB/PBHelperClient.java

```diff
@@ -795,7 +798,12 @@ static public DatanodeInfo convert(DatanodeInfoProto di) {
       long nonDFSUsed = di.getCapacity() - di.getDfsUsed() - di.getRemaining();
       dinfo.setNonDfsUsed(nonDFSUsed < 0 ? 0 : nonDFSUsed);
     }
-    return dinfo.build();
+
+    DatanodeInfo buildInfo = dinfo.build();
+    if (di.hasSoftwareVersion()) {
```

Comment:
How about modifying class `DatanodeInfoBuilder` directly?

Reply:
@zhangshuyan0 Thank you very much for your comment, I will modify it soon.